hub/core/dataset/dataset.py

```diff
@@ -1,6 +1,7 @@
 # type: ignore
 from unittest import skip
 import numpy as np
+import pandas as pd
```

Comment:
this will make pandas a dependancy, lets move it inside the to_dataframe function.